src/opm/parser/eclipse/Deck/DeckKeyword.cpp

```diff
@@ -72,6 +81,12 @@ namespace Opm {
         return m_isDataKeyword;
     }
 
+    const ParserKeyword& DeckKeyword::parserKeyword() const {
+        if (this->parser_keyword)
+            return *this->parser_keyword;
+
+        throw std::logic_error("INTERNAL ERROR: The " + this->m_keywordName + " deck keyword has been instantiated without ParserKeyword information.");
```

Comment:
I think this function is fine, but it is a bit brutal if this is the only way to know if we have a ParserKeyword or not. Adding a `hasParserKeyword()` method would be nicer, but if your intention was that client code should not need to know in the first place, it might be wrong to add it?

Reply:
I am a little hesitant to add that query function. As a general rule the `DeckKeyword` *should have* a valid `ParserKeyword` member. Right now the exceptions are:

1. Some quite special code in the `SummaryConfig`
2. Quite many tests.
3. A temporary `DeckKeyword` is created when an unknown keyword is encountered in the parsing process.

I would prefer to remove all these usecases; but it is not entirely clear to me how to do it - so therefor I have opted to retain the `DeckKeyword(const std::string& )` constructor in this PR - but it should be considered deprecated.